### DISCUSSION TITLE
panic: panic condition sz != 2 occurred when using SE mode

### DISCUSSION THREAD

**Discussion by davidli1515** [gem5/gem5#1293] [Q&A]
Hi guys,
I'm using gem5 for micro-arch security research.
Now I want to run the eviction set building algorithm(https://github.com/TomKessous/Prune-PlumTree---Finding-Eviction-Sets-at-Scale) in SE mode, however this error happens:
src/base/loader/image_file_data.cc:53: panic: panic condition sz != 2 occurred: Couldn't read magic bytes from object file
Memory Usage: 114052 KBytes
Program aborted at tick 0
--- BEGIN LIBC BACKTRACE ---
build/X86/gem5.opt(_ZN4gem515print_backtraceEv+0x30)[0x55d22cb5eb00]
build/X86/gem5.opt(_ZN4gem512abortHandlerEi+0x4e)[0x55d22cb8287e]
/usr/lib/x86_64-linux-gnu/libpthread.so.0(+0x14420)[0x7f8ec351d420]
/usr/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f8ec2ae800b]
/usr/lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f8ec2ac7859]
build/X86/gem5.opt(+0x918ab5)[0x55d22b7e5ab5]
build/X86/gem5.opt(_ZN4gem56loader13ImageFileDataC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0xf69)[0x55d22becbba9]
build/X86/gem5.opt(_ZN4gem56loader16createObjectFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEb+0x46)[0x55d22becc9b6]
build/X86/gem5.opt(+0x1c38495)[0x55d22cb05495]
build/X86/gem5.opt(+0x929f76)[0x55d22b7f6f76]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x14496e)[0x7f8ec367096e]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x27b)[0x7f8ec3669fab]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x5845)[0x7f8ec3665db5]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x15075f)[0x7f8ec367c75f]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x4f23)[0x7f8ec3665493]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x15075f)[0x7f8ec367c75f]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x4f23)[0x7f8ec3665493]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x1e13c2)[0x7f8ec370d3c2]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(PyEval_EvalCode+0x8c)[0x7f8ec370d30c]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x1e8b9d)[0x7f8ec3714b9d]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(+0x144fcb)[0x7f8ec3670fcb]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x334)[0x7f8ec36608a4]
/home/rapiduser/miniconda3/lib/libpython3.10.so.1.0(_PyFunction_Vectorcall+0x78)[0x7f8ec3670dd8]
build/X86/gem5.opt(+0x1c3f874)[0x55d22cb0c874]
build/X86/gem5.opt(main+0x21b)[0x55d22b7ca0fb]
/usr/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f8ec2ac9083]
build/X86/gem5.opt(_start+0x2e)[0x55d22b7d3a8e]
--- END LIBC BACKTRACE ---

FYI, someone has rasied this question(https://www.mail-archive.com/gem5-users@gem5.org/msg20984.html). But he is using FS mode and it's solved(by Jason) by using a correct kernal binary. 
However, I don't think my current guest binary has such problem. (I use the compiled binary called "main" in the given github link:https://github.com/TomKessous/Prune-PlumTree---Finding-Eviction-Sets-at-Scale, and I change all the compiler optimization flags to -static)

Thank you so much for reading my question! Gem5 is struggling!

Best,
David Li

**Comment by darsh612** [comment]
Hey, were you able to solve it ? I am also facing similar issue

**Comment by jefemaestro302** [comment]
Have you had any luck with this problem? I'm also facing it atm
